addons/mail/models/mail_activity.py

```diff
@@ -365,7 +365,7 @@ def _search(self, domain, offset=0, limit=None, order=None):
 
         allowed_ids = defaultdict(set)
         for res_model, res_ids in model_ids.items():
-            records = self.env[res_model].browse(res_ids)
+            records = self.env[res_model].browse(res_ids).exists()
```

Comment:
Would add a small comment :)
```suggestion
            # records may have been removed without their activities (concurrent
            # activity creation / document removal, upgrade, ...) hence adding
            # defensive check
            records = self.env[res_model].browse(res_ids).exists()
```